### ISSUE TITLE
Error 404

### ISSUE THREAD

**Issue by krilson** [WebRTSP/ReStreamer#18]
I tried this out via snap first, a lot of errors about missing plugins (gstreamer and some other non critical). I manually installed gstreamer and the page worked, but the streams did not load, only rotating loading circle indefinitely. 

Attempted to run from source. Compiled everything. ReStreamer runs, no errors visible in the console. But the web page is blank, displaying only "404".

Not sure what else to try. Where is the log stored on the source installed version? Or should I see all the output in the console when running ./ReStreamer?

**Comment by RSATom** [comment]
@krilson what snap channel did you try? (I would recommend you try `edge`)

When it running as standalone app:
* It logs to stdout
* log level can be changed in `~/.config/restreamer.conf`
* you can see GStreamer logs if set something like `GST_DEBUG=*:3`
* it expects `www` in current dir

**Comment by RSATom** [comment]
Also, does `http://ipcam.stream:5080/` works for you?

**Comment by krilson** [comment]
@RSATom, I tried now again from edge. Same behavior as before, page loads, but the loading circle is just spinning. 

Here is the log output: https://pastebin.com/5RSEisZJ

The ERRORs are:
2021-12-28T15:09:41Z rtsp-to-webrtsp.ReStreamer[16772]: 0:00:03.225986475 16772 0x7fb850033e40 ERROR              webrtcbin gstwebrtcbin.c:3540:_set_description_task:<webrtcbin0> media 0 is missing or contains an empty 'ice-ufrag' attribute

It is also complaining about this:
2021-12-28T15:11:11Z rtsp-to-webrtsp.ReStreamer[16772]: Missing decoder: Raw 16-bit PCM audio (audio/x-raw, format=(string)S16LE, layout=(string)interleaved, rate=(int)48000, channels=(int)2, channel-mask=(bitmask)0x0000000000000003)

But there is no audio in this stream, just video.

The install from source on the other hand, gives no output to stdout apart from the notice that server is running...

http://ipcam.stream:5080/ also doesnt work, just spinning loading circle. Are all the streams on this site alive? I tested on a windows machine (chrome and firefox) and on my linux laptop (chrome and firefox).

**Comment by RSATom** [comment]
@krilson I don't have control over all streams except `Bars` and `Record` and the second one almost always not streaming (it's for testing purposes right now). Also `ipcam.stream` vps is on cheap an horrible hosting, so yes, issues are possible. But at least `Bedford Hills` is working for me.

**Comment by RSATom** [comment]
> 2021-12-28T15:09:41Z -[16772]: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

from your log definitely points to some bug in my code...

**Comment by krilson** [comment]
What kind of browser/OS are you using? For me even the `Bedford Hills` doesnt work...

**Comment by RSATom** [comment]
Hm... yes, `Bedford Hills` doesn't work for me anymore too... but `Štrbské pleso` does. I'm on Ubuntu 21 and tried `Chrome Version 95.0.4638.69 (Official Build) (64-bit)` and `Firefox 95.0.1 (64-bit)` with `Štrbské pleso`

But for testing purposes, to avoid third party servers issues, you can try install https://snapcraft.io/rtsp-test-server and point ReStreamer to it.

**Comment by RSATom** [comment]
Such unlucky day... looks like `rtsp-test-server` is broken too :(

**Comment by RSATom** [comment]
No, I was wrong, it's just some compatibility issue with VLC...

**Comment by RSATom** [comment]
I've tried install `ReStreamer` locally with snap, and almost all sources are working with it... It looks like I should change hosting for `ipcam.stream`...

**Comment by krilson** [comment]
I can now see `Western Cape` on your server. So the os/browser is not an issue.

But from my cameras I only get the loading circle. Maybe there is some other issue with the stream on my side. 

Will investigate further. Thanks for all the help.

**Comment by RSATom** [comment]
@krilson Please keep in mind browsers declared to support only `baseline profile` encoded h264 streams. So if your IP Cam doesn't provide supported stream - it will be not shown on browser. But some tricks are possible, for example it's possible to fool browser by provide supported `profile-level-id` in SDP. You can do it by [force-h264-profile-level-id](https://github.com/WebRTSP/ReStreamer/blob/master/restreamer.conf.sample#L21) config option.

Also you can find some more details in browser JS Console.

**Comment by RSATom** [comment]
Closed due to no activity for a long time...
